src/screens/datasharing/SymptomOnsetDateScreen.tsx

```diff
@@ -1,28 +1,38 @@
-import React, {useState} from 'react';
+import React, {useCallback, useContext} from 'react';
 import {ScrollView, StyleSheet} from 'react-native';
 import {Box, Text} from 'components';
 import {useI18n} from 'locale';
+import {useNavigation} from '@react-navigation/native';
+import {FormContext} from 'shared/FormContext';
+import {ContagiousDateType} from 'shared/DataSharing';
 
-import {BaseTekUploadView, DatePicker} from './components';
+import {BaseTekUploadView, DatePicker, StepXofY} from './components';
 
 export const SymptomOnsetDateScreen = () => {
   const i18n = useI18n();
-  const [selectedDate, setSelectedDate] = useState('');
-
+  const navigation = useNavigation();
+  const secondaryButtonOnPress = useCallback(() => navigation.navigate('TekUploadNoDate'), [navigation]);
+  const {data, setSymptomOnsetDate} = useContext(FormContext);
+  const dateParts = data.symptomOnsetDate.split('-');
+  const providedDate = new Date(Number(dateParts[0]), Number(dateParts[1]) - 1, Number(dateParts[2]));
```

Comment:
@smcmurtry  (provided date) thinking we should break this out - and write a unit test.